### ISSUE TITLE
I think I found an unique wayland window id, but of type long.

### ISSUE THREAD

**Issue by rilian-la-te** [lestcape/unity-gtk-module#3]
It will consists of ```PID_MAX_LIMIT*wl_proxy_get_id()+getpid()```
What do you think about it?

**Comment by lestcape** [comment]
It look like all about just the application. Not? How to differentiate windows of the same application?

**Comment by rilian-la-te** [comment]
1.wl_proxy_get_id() is unique for each window (but can be same on another application)
2. getpid() is unique per application

So, a combination of it will be unique, and we will build such combinations, if we multiply wl_proxy_get_id() on PID_MAX_LIMIT(so, it will not mess a pid), and then add a PID to it.

**Comment by lestcape** [comment]
What i want to know now is how will work a QT Wayland window in a Gtk WM and a Gtk Wayland window in a KDE WM base on Qt widgets. Will be 2 WM or they will merged all stuff of Kwin inside Mutter and all stuff of Mutter inside Kwin? Thats will be crazy.

**Comment by rilian-la-te** [comment]
Yes, so, we need a way to identify an active window cross-platform, or at least on each platform.

**Comment by rilian-la-te** [comment]
Identifier and signal will be enough to fix it. I wrote to KDE guys about KDE Wayland Global Menu architecture, may I will take ideas from it.

**Comment by lestcape** [comment]
if wl_proxy_get_id is per windows of application your id it's a valid id. The window xid it' s also a long...I think you can not improve it... But it really will be a long? To many factors... Also if all factors are int32 the result will be a long?

**Comment by rilian-la-te** [comment]
Result will be u_int64_t, which is equal to long on amd64. But if both are int32 on all arches, safer to choose u_int64_t for type of id.

**Comment by rilian-la-te** [comment]
PID is int32, but wl_proxy_get_id() is just int AFAIK, so, it is difflicult to decide.

**Comment by lestcape** [comment]
I think you have an **id**, but probably not in that way... Maybe as an string??

concat(getpid(), "-", wl_proxy_get_id())

**Comment by rilian-la-te** [comment]
I was write you about it, but concat via dot. You do not liked it at past.

**Comment by rilian-la-te** [comment]
so concat(getpid(), ".", wl_proxy_get_id())

**Comment by rilian-la-te** [comment]
String is also not backward-compatible for com.canonical,AppMenu,Registrar.

**Comment by lestcape** [comment]
I can not get the wl_proxy_get_id anyway, so for me it's like nothing... Also structs with pointer are opaque objects and the Gjs introspection can not copy and opaque object if some how you try to shared it with me. Python can copy some opaque objects, probably Vala (the object that are not full-trasnfer), but not gjs.

**Comment by rilian-la-te** [comment]
Vala can do anythink than C can.

**Comment by lestcape** [comment]
Yes, more less... Vala is a rare spawn. Is not considered by some people a real programming language. For me is a programing language, but yes a little out of the standards.
